src/wallet/coinselection.h

```diff
@@ -176,13 +176,13 @@ struct CoinEligibilityFilter
 {
     /** Minimum number of confirmations for outputs that we sent to ourselves.
      * We may use unconfirmed UTXOs sent from ourselves, e.g. change outputs. */
-    const int conf_mine;
+    const int conf_mine{0};
```

Comment:
In f9986bc3ff8f9dc597b5e8b14fceb1df3d473f6e "wallet: CoinEligibilityFilter, set default value for every member"

Is this commit necessary? I don't think we ever create `CoinEligibilityFilter` without using the constructors which initialize these. If the concern is that we might accidentally create a `CoinEligibilityFilter` without using one of those constructors, I would rather that we delete the default constructor rather than setting potentially dangerous default values.

Reply:
Good point and catch.
The only fields that could not be set in the constructors are `max_descendants` and `m_include_partial_groups`. So could add default values for those (my concern was having those fields with garbage), plus remove the default constructor.